src/test/optimism/OptimismOpE2E.t.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import 'forge-std/Test.sol';
+import {AaveV3Optimism} from 'aave-address-book/AaveAddressBook.sol';
+import {GovHelpers} from 'aave-helpers/GovHelpers.sol';
+import {ProtocolV3TestBase, ReserveConfig, ReserveTokens, IERC20} from 'aave-helpers/ProtocolV3TestBase.sol';
+import {BridgeExecutorHelpers} from 'aave-helpers/BridgeExecutorHelpers.sol';
+import {AaveGovernanceV2, IExecutorWithTimelock} from 'aave-address-book/AaveGovernanceV2.sol';
+import {IL2CrossDomainMessenger, AddressAliasHelper} from '../../interfaces/optimism/ICrossDomainMessenger.sol';
+import {IExecutorBase} from '../../interfaces/optimism/IExecutorBase.sol';
+import {CrosschainForwarderOptimism} from '../../contracts/optimism/CrosschainForwarderOptimism.sol';
+import {OpPayload} from '../../contracts/optimism/OpPayload.sol';
+import {DeployL1OptimismProposal} from '../../../script/DeployL1OptimismProposal.s.sol';
+
+contract OptimismOpE2ETest is ProtocolV3TestBase {
+  // the identifiers of the forks
+  uint256 mainnetFork;
+  uint256 optimismFork;
+
+  OpPayload public opPayload;
+
+  address public constant OPTIMISM_BRIDGE_EXECUTOR =
+    0x7d9103572bE58FfE99dc390E8246f02dcAe6f611;
+  IL2CrossDomainMessenger public OVM_L2_CROSS_DOMAIN_MESSENGER =
+    IL2CrossDomainMessenger(0x4200000000000000000000000000000000000007);
+
+  address public constant OP = 0x4200000000000000000000000000000000000042;
+
+  address public constant DAI = 0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1;
+
+  function setUp() public {
+    optimismFork = vm.createFork(vm.rpcUrl('optimism'), 22961333);
+    mainnetFork = vm.createFork(vm.rpcUrl('ethereum'), 15526675);
+    vm.selectFork(optimismFork);
+    vm.startPrank(AaveV3Optimism.ACL_ADMIN);
+    // -------------
+    // Claim pool admin
+    // Only needed for the first proposal on any pool. If ACL_ADMIN was previously set it will ignore
+    // https://github.com/aave/aave-v3-core/blob/master/contracts/dependencies/openzeppelin/contracts/AccessControl.sol#L207
+    // -------------
+    AaveV3Optimism.ACL_MANAGER.addPoolAdmin(OPTIMISM_BRIDGE_EXECUTOR);
+    AaveV3Optimism.ACL_MANAGER.addRiskAdmin(OPTIMISM_BRIDGE_EXECUTOR);
+    vm.stopPrank();
+  }
+
+  function testProposalE2E() public {
+    vm.selectFork(mainnetFork);
+    address crosschainForwarderOptimism = address(
```

Comment:
I think its better to deploy in advance. Is that the intention?

Reply:
intention was to deploy and use the static deployed address one review passed